src/data/markdown/docs/30 xk6-browser/01 Get started/02 Running xk6-browser.md

```diff
@@ -0,0 +1,236 @@
+
+## Asynchronous operations
+
+For some context, k6 operations are synchronous. However, Playwright operations are asynchronous in nature and since xk6-browser aims to provide rough compatibility with Playwright, some of our operations are now asynchronous.
+
+At the moment, `page.waitForNavigation()` and `Element.click()` return promises and work similarly to [JavaScript promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises).
+
+To avoid timing errors or other race conditions in your script, if you have actions that load up a different page, you need to make sure that you wait for that action to finish before continuing.
+
+  <CodeGroup labels={["script.js"]} lineNumbers={[true]}>
+
+  ```javascript
+  import { check } from 'k6';
+  import { chromium } from 'k6/x/browser';
+
+  export default function () {
+    const browser = chromium.launch({ headless: false });
+    const page = browser.newPage();
+
+    page.goto('https://test.k6.io/my_messages.php', { waitUntil: 'networkidle' });
+
+    // Enter login credentials and login
+    page.locator('input[name="login"]').type('admin');
+    page.$('input[name="password"]').type('123');
+
+    // Wait for asynchronous operations
+    // eslint-disable-next-line no-undef
+    Promise.all([page.waitForNavigation(), page.$('input[type="submit"]').click()])
+      .then(() => {
+        check(page, {
+          header: page.locator('h2').textContent() == 'Welcome, admin!',
+        });
+      })
+      .finally(() => {
+        page.close();
+        browser.close();
+      });
+  }
+  ```
+
+  </CodeGroup>
+
+The preceding code uses `Promise.all([])` to wait for the two promises to be resolved before continuing. Then, you can use [`check`](https://k6.io/docs/javascript-api/k6/check/) from the k6 API to assert the text content of a specific element. Finally, you close the page and the browser.
+
+## Assertions
+
+You can add assertions in your browser script via:
+1. Checks, as mentioned in the previous section
+2. [k6chaijs](https://k6.io/docs/javascript-api/jslib/k6chaijs/) for a more BDD (Behavior Driven Development) or TDD (Test Driven Development) style
+
+  <CodeGroup labels={["script.js"]} lineNumbers={[true]}>
+
+  ```javascript
+  import { chromium } from 'k6/x/browser';
+  import { expect } from 'https://jslib.k6.io/k6chaijs/4.3.4.0/index.js';
+
+  export default function () {
+    const browser = chromium.launch({ headless: false });
+    const page = browser.newPage();
+
+    page.goto('https://test.k6.io/my_messages.php', { waitUntil: 'networkidle' });
+
+    // Enter login credentials and login
+    page.locator('input[name="login"]').type('admin');
+    page.$('input[name="password"]').type('123');
+
+    // Wait for asynchronous operations
+    // eslint-disable-next-line no-undef
```

Comment:
IMO, this comment can confuse users. Should we explain it or write `Promise.all` differently?

Reply:
`<!-- eslint-skip -->` works now so I've remove the eslint single line comment :)